src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableQueue_1.cs

```diff
@@ -76,7 +75,11 @@ public ImmutableQueue<T> Clear()
         /// </value>
         public bool IsEmpty
         {
-            get { return _forwards.IsEmpty && _backwards.IsEmpty; }
+            get
+            {
+                Debug.Assert(!_forwards.IsEmpty || _backwards.IsEmpty);
```

Comment:
I removed the `&& _backwards.IsEmpty` because AFAICS if the forwards stack is empty then the backwards is, too. I added this assert to ensure that.